js/ui/tab_cookies.js

```diff
@@ -371,19 +419,54 @@
     };
   }
 
-  function createDomainObj(domain, cookienum)
+  function createDomainObj(domain, cookienum, whitelist = false)
   {
     return {
       dataset: {
-        access: domain
+        access: domain,
+        whitelist: whitelist
       },
       texts: {
         domain: domain,
         cookienum: cookienum + " Cookies"
       }
     };
   }
-
+  function updateWhitelistInList(change)
+  {
+    if ("cookieWhitelist" in change) 
+    {
+      let newValue = change.cookieWhitelist.newValue;
+      for (var domain in newValue)
```

Comment:
Suggestion: I can imagine that iterating over whole table and updating items using Browser API can have performance implications, what we can do here is to compare old and new values and act accordingly instead of querying each item.

Let me know please if you want me to elaborate on this with code examples.

Reply:
There is two solutions I see.
Determine difference and act:
```
onStorageChange.addListener(function(changeInfo)
{
  if ("cookieWhitelist" in change)
  {
    let {newValue, oldValue} = change.cookieWhitelist;
    let domains = compare(newValue, oldValue);
    updateWhitelistingList(domains);
  }
});
```

Comparing when requesting items from memory `tableList.getItem(domain);`:

```
let domainItem = tableList.getItem(domain);

if (domainItem && domainItem. dataset.whitelist == newValue[domain].domainWhitelisted)
{
...
}
```

Reply:
added some code to filter out unchanged domains (domainWhitelist and cookies both unchanged) before calling updateWhitelistInList
also updated getcookies to only send current domain to updateWhitelistInList
so the only time updateWhitelistInList should go through everything is if the cookies tab is filled for the first time